From f9d8d0e7e1385658a112c441e1eb7ff165f3fbce Mon Sep 17 00:00:00 2001
From: Andrey Smirnov <andrew.smirnov@gmail.com>
Date: Tue, 18 Jan 2022 08:44:34 -0800
Subject: [PATCH 17/21] usb: dwc3: Bump USB gadget wakeup timeout

Seems to be necessary to avoid timouts when switching host/gadged on
Steam Deck.

(cherry picked from commit b69aafb85c7794a729e68e92214e539e7e820192)
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 drivers/usb/dwc3/gadget.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 31a654c6f15b..99d736aa0194 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2409,7 +2409,7 @@ static int __dwc3_gadget_wakeup(struct dwc3 *dwc, bool async)
 		return 0;
 
 	/* poll until Link State changes to ON */
-	retries = 20000;
+	retries = 2000000;
 
 	while (retries--) {
 		reg = dwc3_readl(dwc->regs, DWC3_DSTS);
-- 
2.48.1

